tensorflow/lite/kernels/internal/reference/div.h

```diff
@@ -172,11 +212,92 @@ inline void BroadcastDivSlow(const ArithmeticParams& params,
                              const int8_t* input2_data,
                              const RuntimeShape& unextended_output_shape,
                              int8_t* output_data) {
-  BroadcastDivSlowQuantized<int8_t, N>(
-      params, unextended_input1_shape, input1_data, unextended_input2_shape,
-      input2_data, unextended_output_shape, output_data);
+  if (!params.is_quantized) {
+    int8_t output_activation_min;
+    int8_t output_activation_max;
+    GetActivationParams(params, &output_activation_min, &output_activation_max);
+
+    TFLITE_DCHECK_LE(unextended_input1_shape.DimensionsCount(), N);
+    TFLITE_DCHECK_LE(unextended_input2_shape.DimensionsCount(), N);
+    TFLITE_DCHECK_LE(unextended_output_shape.DimensionsCount(), N);
+
+    NdArrayDesc<N> desc1;
+    NdArrayDesc<N> desc2;
+    NdArrayDesc<N> output_desc;
+    NdArrayDescsForElementwiseBroadcast(
+        unextended_input1_shape, unextended_input2_shape, &desc1, &desc2);
+    CopyDimsToDesc(RuntimeShape::ExtendedShape(N, unextended_output_shape),
+                   &output_desc);
+
+    // In Tensorflow, the dimensions are canonically named (batch_number, row,
+    // col, channel), with extents (batches, height, width, depth), with the
+    // trailing dimension changing most rapidly (channels has the smallest
+    // stride, typically 1 element).
+    //
+    // In generated C code, we store arrays with the dimensions reversed. The
+    // first dimension has smallest stride.
+
+    auto div_func = [&](int indexes[N]) {
+      output_data[SubscriptToIndex(output_desc, indexes)] =
+          ActivationFunctionWithMinMax<int8_t>(
+              input1_data[SubscriptToIndex(desc1, indexes)] /
+                  input2_data[SubscriptToIndex(desc2, indexes)],
+              output_activation_min, output_activation_max);
+    };
+    NDOpsHelper<N>(output_desc, div_func);
+  } else {
+    BroadcastDivSlowQuantized<int8_t, N>(
+        params, unextended_input1_shape, input1_data, unextended_input2_shape,
+        input2_data, unextended_output_shape, output_data);
+  }
 }
+template <int N = 5>
+inline void BroadcastDivSlow(const ArithmeticParams& params,
+                             const RuntimeShape& unextended_input1_shape,
+                             const int16_t* input1_data,
+                             const RuntimeShape& unextended_input2_shape,
+                             const int16_t* input2_data,
+                             const RuntimeShape& unextended_output_shape,
+                             int16_t* output_data) {
+  if (!params.is_quantized) {
```

Comment:
Why no use the existing templated `template <typename T, int N = 5> void BroadcastDivSlow` function for the unquantized path?